hooks/checkout

```diff
@@ -13,39 +25,62 @@ check_set() {
   fi
 }
 
+# Checks for failed exit codes returned by a timeout command.
+# If the underlying command fails due to timeout, this function stops the current execution returning either the
+# TIMEOUT exit code (124) or the override value provided by the caller.
+# In any other non-timeout failure scenarios, the current execution is stopped but the underlying command's exit
+# code is returned unmodified.
+# Arguments:
+#   $1: The exit code returned by the timeout command.
+#   $2: The exit code to be returned when the underlying command times out.
+check_timeout_exit_code() {
+  local exit_code="$1"
+  local timeout_exit_code_override="${2:-}"
```

Comment:
This doesn't need to be a parameter, it's already available in scope as `GIT_REMOTE_TIMEOUT_EXIT_CODE`, and it's always set. So you don't need the `-n` check either.

Reply:
Well, I did it this way so that we could reuse this function if we ever use `timeout` for any other commands (i.e S3).

Reply:
Ah yes, that makes sense, but let's add that complexity when we need it.